src/approval/approval.service.ts

```diff
@@ -0,0 +1,96 @@
+import Axios, { AxiosResponse, AxiosInstance } from 'axios';
+import { IApproverInfo, ICanApproveToUser, IApprovalRequest } from './approvers.interface';
+import { ApprovalError, NotFoundError, ApplicationError, UnauthorizedError } from '../utils/errors/errors';
+import { config } from '../config';
+import { getToken } from "../spike/spike.service";
+import { TransferRepository } from '../transfer/transfer.repository';
+
+export class ApprovalService {
```

Comment:
A proposal:
You could create an abstract class that this class extends which handles the integration with the spike-service, and maybe also wrap the function with APM spans for some nice monitoring.